### ISSUE TITLE
[BUG] Don't downgrade Numpy

### ISSUE THREAD

**Issue by Kochise** [sinanislekdemir/payton#31]
Spyder 5.0.5 : https://github.com/spyder-ide/spyder/releases/

``` batch
C:\Program Files\Spyder\Python>python -m pip install --upgrade payton
Collecting payton
  Downloading Payton-0.1.5-py3-none-any.whl (113 kB)
     |████████████████████████████████| 113 kB 3.3 MB/s
Collecting PyOpenGL-accelerate==3.1.5
  Downloading PyOpenGL_accelerate-3.1.5-cp37-cp37m-win_amd64.whl (329 kB)
     |████████████████████████████████| 329 kB 6.8 MB/s
Collecting numpy==1.18.2
  Downloading numpy-1.18.2-cp37-cp37m-win_amd64.whl (12.8 MB)
     |████████████████████████████████| 12.8 MB 6.8 MB/s
Collecting Pillow==8.1.1
  Downloading Pillow-8.1.1-cp37-cp37m-win_amd64.whl (2.2 MB)
     |████████████████████████████████| 2.2 MB 6.4 MB/s
Collecting PySDL2==0.9.7
  Downloading PySDL2-0.9.7-py3-none-any.whl (541 kB)
     |████████████████████████████████| 541 kB 6.4 MB/s
Collecting PyOpenGL==3.1.3rc1
  Downloading PyOpenGL-3.1.3rc1-py3-none-any.whl (2.4 MB)
     |████████████████████████████████| 2.4 MB 6.4 MB/s
Collecting pyrr==0.10.3
  Downloading pyrr-0.10.3-py3-none-any.whl (46 kB)
     |████████████████████████████████| 46 kB 470 kB/s
Collecting multipledispatch
  Downloading multipledispatch-0.6.0-py3-none-any.whl (11 kB)
Requirement already satisfied: six in c:\program files\spyder\pkgs (from multipledispatch->pyrr==0.10.3->payton) (1.16.0)
Installing collected packages: numpy, multipledispatch, PySDL2, pyrr, PyOpenGL-accelerate, PyOpenGL, Pillow, payton
  Attempting uninstall: numpy
    **Found existing installation: numpy 1.19.3
    Uninstalling numpy-1.19.3:
      Successfully uninstalled numpy-1.19.3**
  Attempting uninstall: Pillow
    Found existing installation: Pillow 8.2.0
    Uninstalling Pillow-8.2.0:
      Successfully uninstalled Pillow-8.2.0
Successfully installed Pillow-8.1.1 PyOpenGL-3.1.3rc1 PyOpenGL-accelerate-3.1.5 PySDL2-0.9.7 multipledispatch-0.6.0 numpy-1.18.2 payton-0.1.5 pyrr-0.10.3

C:\Program Files\Spyder\Python>
```

Uninstalling numpy-1.19.3
Installing numpy-1.18.2

What is the point ?

Be less specific in your dependency check :

requirements.txt
```
numpy==1.18.2
Pillow==8.2.0
PyOpenGL==3.1.3rc1
pyrr==0.10.3
PySDL2==0.9.7
PyOpenGL-accelerate==3.1.5
```

Don't use `pip freeze > requirements.txt` straight out the box, adapt it :

```
numpy>=1.18.2
Pillow>=8.2.0
PyOpenGL>=3.1.3rc1
pyrr>=0.10.3
PySDL2>=0.9.7
PyOpenGL-accelerate>=3.1.5
```

See https://pip.pypa.io/en/stable/user_guide/#requirements-files

**Comment by sinanislekdemir** [comment]
Hi @Kochise 
Thank you so much for pointing out that issue. Apologies for the inconvenience as well. 
I fixed the issue with: 19a7dc786b6a722623929fbe954dc269eca66598 and released `0.1.6` to pypi already. 

Also this made me see that numpy finally made some huge improvements with typehints and as a result Payton's internal types require some minor refactoring but not a blocker at this state. I will work on them during my upcoming vacation :)

Have a nice weekend!

**Comment by sinanislekdemir** [comment]
Okay, I couldn't wait for my vacation to fix typehint problems with Numpy. (c6e70939537fd552459e2327a30b2991025cad50)

I should say that, some parts of it was tricky. Not blaming but some guys at Numpy community are too lazy to write `Any` at everywhere 😅 

Version `0.1.7` is out and has a minor fix + better typehint support.
